src/Network/Algorand/Node/Util.hs

```diff
@@ -29,3 +40,22 @@ transactionStatus TransactionInfo{tiConfirmedRound, tiPoolError} =
     Nothing -> case T.null tiPoolError of
       False -> KickedOut tiPoolError
       True -> Waiting
+
+-- | Catches error that has block absence for the round as the cause and
+-- return Nothing in this case. Other errors would be rethrown.
+getBlock
+  :: MonadCatch m
+  => Api.ApiV2 (AsClientT m) -> B.Round -> m (Maybe B.Block)
+getBlock api rnd = handle handler $ do
+  B.BlockWrapped block <- Api._block api rnd Api.msgPackFormat
+  pure (Just block)
+  where
+    noBlockMsg = "ledger does not have entry"
```

Comment:
Surprisingly, when I call `https://api.testnet.algoexplorer.io/v2/blocks/1523607922`, I get
```
{
    "message": "failed to retrieve information from the ledger"
}
```

Am I calling wrong endpoint? Could you please check that it still returns this message?

Reply:
If I change this message to `"failed to retrieve information from the ledger"`, then:

```
λ stack exec -- halgo node block --round 1523607922                                                                                                          0 
Response 500 (Internal Server Error): {"message":"ledger does not have entry 1523607922 (latest 15629174, committed 15629174)"}
```

with current code:
```
λ stack exec -- halgo node block --round 1523607922                                                                                                          0 
No block found
```

Reply:
Maybe it's because we send requests in msg pack format but in browser you can see json response and they have different messages for those formats

Reply:
![image](https://user-images.githubusercontent.com/2556409/126997500-2ad6bff5-5d4a-4f39-9c0b-41c6923e3c99.png)

Oh, I am even more surprised, but it turns out you are right. So the error message really depends on `format`. Then it's even worse for us, because in order to support `format=json` we'll need to update this function.